File: Server/msgTypes/PlaneCalculatedTrajectoryPoints.cs
public class PlaneCalculatedTrajectoryPoints
{
    public string planeName { get; set; }
    public List<TrajectoryPoint> trajectoryPoints { get; set; }
    public PlaneCalculatedTrajectoryPoints()
    {

    }
    public PlaneCalculatedTrajectoryPoints(string planeName, List<TrajectoryPoint> trajectoryPoints)
    {
        this.planeName = planeName;
        this.trajectoryPoints = trajectoryPoints;
    }
    public override string ToString()
{
    return $"PlaneName: {planeName}, TrajectoryPoints Count: {trajectoryPoints?.Count ?? 0}";
}
}

File: Client/src/js/msgTypes/PlaneCalculatedTrajectoryPoints.js
import { TrajectoryPoint } from "./TrajectoryPoint.js";
import { GeoPoint } from "./GeoPoint.js";
export class PlaneCalculatedTrajectoryPoints {
    constructor(raw) {
        this.planeName = raw.planeName;
        this.trajectoryPoints = raw.trajectoryPoints.map(tp =>
            new TrajectoryPoint(
                new GeoPoint(tp.position.longitude, tp.position.latitude, tp.position.altitude),
                tp.heading,
                tp.pitch
            )
        );
    }
    toString() {
        const pointsStr = this.trajectoryPoints.map(tp => tp.toString()).join(", ");
        return `Plane: ${this.planeName}\nTrajectoryPoints: [${pointsStr}]`;
    }
}